src/main/kotlin/JobState.kt

```diff
@@ -53,6 +66,10 @@ class JobState<U: Any> constructor(val value: U?, val interceptors: List<WorkerI
     suspend fun executeToResult(fn: suspend (U) -> WorkerResult) = processWorkerFunction(fn)
     suspend fun executeToResult(msg: String, fn: suspend (U) -> WorkerResult) = processWorkerFunction(fn, msg)
 
+
+    suspend fun <R: Any> executeToEither(fn: suspend (U) -> Either<R>): JobState<R> = processEither(fn)
```

Comment:
The operation for some monad M<T> taking function F: T -> M<U>  returning M<U> is usually called a `flatMap`. We haven't followed this convention in other cases so maybe it is not appropriate to use it here. `executeTo$NEW_NAME_OF_EITHER` is fine